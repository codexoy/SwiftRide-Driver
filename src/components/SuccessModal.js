import React from "react"
import { StyleSheet, View, Text, Modal, TouchableOpacity } from "react-native"
import FastImage from "react-native-fast-image"
import { floatButtonOnlyStyle, MAIN_PADDING } from "../consts"
const images = {
	lesson: require("../../assets/images/lesson_success.png"),
	payment: require("../../assets/images/payment_success.png"),
	signup: require("../../assets/images/sign_up_success.png")
}
export default class SuccessModal extends React.Component {
	render() {
		return (
			<Modal
				animationType="slide"
				transparent={false}
				visible={this.props.visible}
				onRequestClose={() => {}}
			>
				<View style={styles.container}>
					<FastImage
						style={styles.img}
						source={images[this.props.image]}
						resizeMode={FastImage.resizeMode.contain}
					/>
					<Text style={styles.title}>{this.props.title}</Text>
					<Text style={styles.desc}>{this.props.desc}</Text>
					<TouchableOpacity
						style={styles.button}
						onPress={() => {
							this.setState({ visible: false }, () => {
								this.props.buttonPress()
							})
						}}
					>
						<Text style={styles.btnText}>{this.props.button}</Text>
					</TouchableOpacity>
				</View>
			</Modal>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center",
		alignItems: "center",
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING
	},
	img: {
		width: 120,
		height: 120
	},
	title: {
		fontSize: 24,
		fontWeight: "bold",
		marginTop: 24
	},
	desc: {
		color: "gray",
		marginTop: 12,
		textAlign: "center"
	},
	button: {
		...floatButtonOnlyStyle,
		marginTop: 40
	},
	btnText: {
		color: "#fff"
	}
})
