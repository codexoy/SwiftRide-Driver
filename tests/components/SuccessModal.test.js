import React from "react"

import renderer from "react-test-renderer"

import SuccessModal from "../../src/components/SuccessModal"

describe("SuccessModal", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<SuccessModal
					visible={true}
					image="lesson"
					title="test"
					desc="desc test"
					button="button test"
					buttonPress={jest.fn}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
