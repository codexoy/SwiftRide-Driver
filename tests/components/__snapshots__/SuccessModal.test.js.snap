// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal view renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 26,
        "paddingRight": 26,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 120,
            "width": 120,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../assets/images/lesson_success.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginTop": 24,
        }
      }
    >
      test
    </Text>
    <Text
      style={
        Object {
          "color": "gray",
          "marginTop": 12,
          "textAlign": "center",
        }
      }
    >
      desc test
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(12,116,244)",
          "borderRadius": 28,
          "elevation": 8,
          "height": 56,
          "justifyContent": "center",
          "marginTop": 40,
          "opacity": 1,
          "shadowColor": "rgb(12,116,244)",
          "shadowOffset": Object {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 16,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        button test
      </Text>
    </View>
  </View>
</Modal>
`;
